jobs/build/update-base-images/Jenkinsfile

```diff
@@ -0,0 +1,60 @@
+node {
+    checkout scm
+    def commonlib = load("pipeline-scripts/commonlib.groovy")
+    def buildlib = load("pipeline-scripts/buildlib.groovy")
+    // Expose properties for a parameterized build
+    properties(
+        [
+            buildDiscarder(
+                logRotator(
+                    artifactDaysToKeepStr: '',
+                    artifactNumToKeepStr: '',
+                    daysToKeepStr: '',
+                    numToKeepStr: '100')
+                ),
+                [
+                $class: 'ParametersDefinitionProperty',
+                parameterDefinitions: [
+                    [
+                        name: 'BASE_IMAGE_TAGS',
+                        description: 'list of openshift cve base images.',
+                        $class: 'hudson.model.StringParameterDefinition',
+                        defaultValue: commonlib.ocpBaseImages.join(',')
+                    ],
+                    [
+                        name: 'MAIL_LIST_FAILURE',
+                        description: 'Failure Mailing List',
+                        $class: 'hudson.model.StringParameterDefinition',
+                        defaultValue: [
+                            'aos-team-art@redhat.com',
+                        ].join(',')
+                    ],
+                    commonlib.mockParam(),
+                ]
+            ],
+            disableConcurrentBuilds()
+        ]
+    )
+
+    commonlib.checkMock()
+    buildlib.initialize()
+    def imageName = params.BASE_IMAGE_TAGS.split(',')
+
+
+    currentBuild.displayName = "#${currentBuild.number} Update base images"
+    currentBuild.description = ""
+    try {
+        imageName.each { image ->
```

Comment:
unfortunately, while groovy has this feature, jenkins won't let you use it. you get the dreaded:
```
an exception which occurred:
	in object org.jenkinsci.plugins.workflow.cps.CpsThreadGroup@33db9140
Caused: java.io.NotSerializableException: java.util.ArrayList$Itr
```
This is because everything you use needs to be either serializable or in a @NonCPS-annotated method, and iterators aren't serializable, and `sh` can't go in NonCPS.

This is why you see so much code using the `for(i=1;...;...)` syntax. Unfortunately it's necessary.